app/controllers/posts_controller.rb

```diff
@@ -25,4 +25,23 @@ def show
 
     render 'show'
   end
+
+  def new
+    @id = current_user.id
+    render 'new'
+  end
+
+  def create
+    title = params[:post]['title']
+    text = params[:post]['text']
+    post = Post.new(user_id: current_user.id, title:, text:, comments_counter: 0, likes_counter: 0)
+
+    if post.save
+      flash[:success] = 'Post saved successfully'
+      redirect_to "/users/#{current_user.id}/posts"
+    else
+      flash.now[:error] = 'Error: Question could not be saved'
+      render 'new'
+    end
+  end
 end
```

Comment:
- [ ] [OPTIONAL] I would like to recommend you to add a **params method** because with params  inside your controller action's you can call params to access form & URL query data. Please follow [this documentation](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides) to know more about :+1: